test/jdk/tools/launcher/TestXcheckJNIWarnings.java

```diff
@@ -25,14 +25,22 @@
 import java.io.IOException;
 
 /**
- * @test
+ * @test id=noagent
  * @bug 8187442
  * @summary Launching app shouldn't produce any jni warnings.
  * @modules jdk.compiler
  *          jdk.zipfs
- * @compile TestXcheckJNIWarnings.java
  * @run main TestXcheckJNIWarnings
  */
+
+/**
+ * @test id=jdwp-agent
+ * @bug 8187442
+ * @summary Launching app with jdwp agent shouldn't produce any jni warnings.
+ * @modules jdk.compiler
+ *          jdk.zipfs
+ * @run main TestXcheckJNIWarnings -agentlib:jdwp=transport=dt_socket,server=y,suspend=n
```

Comment:
This needed to be `main/othervm`. Not even sure how this can work unless in othervm mode!

Reply:
Sorry ignore that - I see now.